framework/decode/vulkan_replay_consumer_base.h

```diff
@@ -1203,6 +1216,26 @@ class VulkanReplayConsumerBase : public VulkanConsumer
     std::unordered_set<uint32_t>      removed_swapchain_indices_;
     std::vector<uint32_t>             capture_image_indices_;
     std::vector<SwapchainKHRInfo*>    swapchain_infos_;
+
+  protected:
+    // Used by pipeline cache handling, there are the following two cases for the flag to be set:
+    //
+    //    1. Replay with command line option --opcd or --omit-pipeline-cache-data and some
+    //       pipeline cache data was really omitted.
+    //
+    //    2. Replay without command line option --opcd or --omit-pipeline-cache-data and there is
+    //       at least one vkCreatePipelineCache call with valid initial pipeline cache data and
+    //       the initial cache data has no corresponding replay time cache data.
+    bool omitted_pipeline_cache_data_;
```

Comment:
There are the following cases in OverrideCreatePipelineCache:

#User start replay with omit_pipeline_cache_data option

    ##A> options_.omit_pipeline_cache_data and pipeline cache created with initialDataSize==0

    ##B> options_.omit_pipeline_cache_data and pipeline cache created with initialDataSize!=0, omitted_pipeline_cache_data_ set true 

#User start replay without omit_pipeline_cache_data option

    ##C> !options_.omit_pipeline_cache_data and pipeline cache created with initialDataSize==0

    ##D> !options_.omit_pipeline_cache_data and pipeline cache created with initialDataSize!=0 and pInitialData can be mapped by this PR

    ##E> !options_.omit_pipeline_cache_data and pipeline cache created with initialDataSize!=0 and pInitialData can not be mapped by this PR, omitted_pipeline_cache_data_ set true 

In the handling of this PR, once any vkCreatePipelineCache command of B> or E> show up, for all following pipeline creation, the handling of AllowCompileDuringPipelineCreation will be added.

Compared with every pipeline cache has different operation for AllowCompileDuringPipelineCreation, we saved the cache usage tracking handling (such tracking need consider merging cache), and the running difference is:

   AllowCompileDuringPipelineCreation was added to some pipeline creation call which is not really needed, this includes those pipeline cache which is successfully mapped.

I think we don't need to care of the difference. The reason being: 

Suppose a pipeline creation call X, its pipeline cache is successfully mapped (we don't need to consider map failed, because for the case we must add AllowCompileDuringPipelineCreation, otherwise playback will fail), we can see creation compile operation is same if compare the pipeline creation with or without AllowCompileDuringPipelineCreation:

 1> Pipeline creation with handling of AllowCompileDuringPipelineCreation, the command will proceed the pipeline creation without re-compile.
 2> Pipeline creation without handling of AllowCompileDuringPipelineCreation, the command will proceed the pipeline creation without re-compile.